src/formats/cif.cpp

```diff
@@ -0,0 +1,205 @@
+        int model = std::stoul(row[11]);
+        const std::string &chain = row[1];
+        const std::string &residue = row[4];
+        const std::string &res_num = row[2];
+        const std::string atom = row[5][0] != '"' ? row[5] : row[5].substr(1, row[5].size() - 2);
+
+        if (_model == model) {
+            if (_chain == chain) {
+                if (_res_num == res_num && _residue == residue) {
+                    if (_atom == atom) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+};
+
+
+void CIF::write_cif_block(std::ostream &out,
+                          gemmi::cif::Table &table, 
+                          std::vector<std::string> &p_charge, 
+                          std::vector<std::string> &vdw_radii) {
+
+    auto &loop = *table.get_loop();
+
+    unsigned long orig_tag_size = loop.tags.size();
+    unsigned long new_tag_size = orig_tag_size + 2;
+
+    // Creates a new table full of empty strings with the correct number of dimensions
+    // Outside vector size is the # of columns, inside vector size is the # of rows.
+    std::vector<std::vector<std::string>> newCols(new_tag_size, {loop.length(), {"Empty"}});
+
+    // Copies data from original columns to their respecitve column in the new table filled with empty strings.
+    // Leaving only the new appended columns as empty strings
+    for (unsigned int i = 0; i != orig_tag_size; ++i) {
+        auto iCol = table.bloc.find_loop(loop.tags[i]);
+        std::copy(iCol.begin(), iCol.end(), newCols[i].begin());
+    }
+
+    newCols[new_tag_size - 2] = std::move(p_charge);
+    newCols[new_tag_size - 1] = std::move(vdw_radii);
+
+    std::vector<std::string> new_tags{
+        "_atom_site.fw2_charge",
+        "_atom_site.fw2_vdw_radius"};
+    for (const auto &tag : new_tags)
+        loop.tags.push_back(tag);
+
+    loop.set_all_values(newCols);
+
+    gemmi::cif::write_cif_block_to_stream(out, table.bloc);
+}
+
+
+void CIF::save_charges(const MoleculeSet &ms, const Charges &charges, const std::string &filename) {
+
+    fs::path out_dir{config::chg_out_dir};
+    std::string out_filename = fs::path(filename).filename().replace_extension(".fw2.cif").string();
+    std::string out_file{(out_dir / out_filename).string()};
+    std::ofstream out_stream{out_file};
+
+    auto doc = gemmi::cif::read_file(filename);
+    auto& block = doc.sole_block();
+    auto table = block.find("_atom_site.", atom_site_columns);
+
+    if (!table.ok()) {
+        fmt::print(stderr, "_atom_site category is empty for file: {}\n", filename);
+        exit(EXIT_FILE_ERROR);
+    }
+
+    auto p_charge  = std::vector<std::string>{table.length(), "?"};
+    auto vdw_radii = std::vector<std::string>{table.length(), "?"};
+
+    const auto &molecule = ms.molecules()[0];
+
+
+    // ChargeFW2 is hardcoded to only read first model.
+    const int model{1};
```

Comment:
Please use initialization with = for these simple variables.

Reply:
sure.